demos/speech_recognition_demo/python/speech_recognition_demo.py

```diff
@@ -55,56 +65,74 @@ def get_profile(profile_name):
-        audio = np.frombuffer(wave_read.readframes(pcm_length * channel_num), dtype=np.int16).reshape((pcm_length, channel_num))
-        wave_read.close()
-    print("Loading, including network weights, IE initialization, LM, building LM vocabulary trie, loading audio: {} s".format(timer.elapsed))
-    print("Audio file length: {} s".format(audio.shape[0] / sampling_rate))
-
-    # Now it is enough to call:
-    #   transcription = stt.recognize_audio(audio, sampling_rate)
-    # if you don't need to access intermediate features like character probabilities or audio features.
-
-    with Timer() as timer:
-        audio_features = stt.extract_mfcc(audio, sampling_rate=sampling_rate)
-    print("MFCC time: {} s".format(timer.elapsed))
-
-    with Timer() as timer:
-        character_probs = stt.extract_per_frame_probs(audio_features, wrap_iterator=tqdm)
-    print("RNN time: {} s".format(timer.elapsed))
-
-    with Timer() as timer:
-        transcription = stt.decode_probs(character_probs)
-    print("Beam search time: {} s".format(timer.elapsed))
-    print("Overall time: {} s".format(timeit.default_timer() - start_time))
-
-    print("\nTranscription and confidence score:")
+        print("Audio file length: {} s".format(pcm_length / sampling_rate))
+        print("")
+
+        audio_pos = 0
+        play_start_time = timeit.default_timer()
+        iter_wrapper = tqdm if not args.online else (lambda x: x)
+        for audio_iter in iter_wrapper(range(0, pcm_length, args.block_size)):
+            audio_block = np.frombuffer(wave_read.readframes(args.block_size * channel_num), dtype=np.int16).reshape((-1, channel_num))
+            if audio_block.shape[0] == 0:
+                break
+            audio_pos += audio_block.shape[0]
+            #
+            # It is possible to call stt.recognize_audio(): 1) for either whole audio files or
+            # by splitting files into blocks, and 2) to reuse stt object for multiple files like this:
+            #   transcription1 = stt.recognize_audio(whole_audio1, sampling_rate)
+            #   transcription2 = stt.recognize_audio(whole_audio2, sampling_rate)
+            #   stt.recognize_audio(whole_audio3_block1, sampling_rate, finish=False)
+            #   transcription3 = stt.recognize_audio(whole_audio3_block2, sampling_rate, finish=True)
+            # If you need intermediate features, you can call pipeline stage by stage like this:
+            #    audio_features = stt.extract_mfcc(whole_audio, sampling_rate)
+            #    character_probs = stt.extract_per_frame_probs(audio_features)
+            #    transcription = stt.decode_probs(character_probs)
+            #
+            partial_transcr = stt.recognize_audio(audio_block, sampling_rate, finish=False)
+            if args.online:
+                if partial_transcr is not None and len(partial_transcr) > 0:
+                    print('\r' + partial_transcr[0]['text'][-args.online_window:], end='')
```

Comment:
```suggestion
                    print('\r' + partial_transcr[0].text[-args.online_window:], end='')
```

Reply:
Done. This bug was added when I rebased my branch. So I fixed it in the commit that adds this code (when I was rebasing another time), not in the last commit containing the other changes -- to keep the state between commits working.